pkgs/by-name/s7/s7/package.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  notcurses,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "s7";
+  version = "0-unstable-2024-09-02";
+
+  src = fetchFromGitLab {
+    domain = "cm-gitlab.stanford.edu";
+    owner = "bil";
+    repo = "s7";
+    rev = "c10d3742d263276d8934af8baaa379742bd63c35";
+    hash = "sha256-cEXveNW2bl3jj21axEAC2vOzMHDH+tA1cpTcpT5NySg=";
+  };
+
+  nativeBuildInputs = [
+    notcurses
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    gcc s7.c \
+        -o s7-nrepl \
+        -O2 \
+        -I. \
+        -Wl,-export-dynamic \
+        -lm \
+        -ldl \
+        -DWITH_MAIN \
+        -DWITH_NOTCURSES \
+        -lnotcurses-core
+
+    runHook postBuild
+  '';
```

Comment:
this is really cool! I never heard of s7 before. From the docs it seems that the recommended way to actually use this is to include s7.c and s7.h in your final application, rather than link it. In that sense does it even make sense to build it here? Maybe my question is: what is this actually building, if the point is just to include it as source?

And there seems to be a check file included, s7test.scm -- how much work would it be to compile and run that on doCheck ?

Reply:
> what is this actually building, if the point is just to include it as source

You can build s7 as a standalone REPL, which is probably useful at least as an evaluation tool to check the language feature support quickly (and eventually someone will like this specific Scheme implementation best as a standalone REPL too)

Reply:
It builds `nrepl` that is embedded in the upstream code.  (sorry I'm not able to build the older `repl`.) The REPL is optional, indeed, but it is so feature-rich that it looks like more a Emacs-style TUI text editor rather than other Scheme REPLs, some of which can't even handle arrow keys. It would be absolutely a pity if the interactive environment just sank into a code tree of someone else and did not show up as itself.